packages/web-pkg/src/composables/pagination/usePagination.ts

```diff
@@ -40,6 +45,32 @@ export function usePagination<T>(options: PaginationOptions<T>): PaginationResul
     return unref(options.items).slice(start, end)
   })
 
+  eventBus.subscribe(
+    'app.files.navigate.page',
+    async ({ resourceId, forceScroll }: { resourceId: string; forceScroll: boolean }) => {
+      const index = findIndex(unref(options.items), (item: any) => item.id === resourceId)
+      if (index >= 0) {
+        const page = Math.ceil((index + 1) / Number(unref(perPage)))
+        await router.push({ ...unref(route), query: { ...unref(route).query, page } })
+
```

Comment:
Looking at this again, I'm thinking maybe we should actually send a `app.files.navigate.scroll` event (or similar) over the `eventBus`here. Otherwise it feels too tightly coupled to the dom structure.

```suggestion
    }
    eventBus.publish('app.files.navigate.scrollTo', { resourceId, forceScroll })
  }
```


I think I like that better than using another instance of the useScrollTo composable with an `doActualScroll` method...


@JammingBen thoughts?

Reply:
I agree with not handling the scroll logic here. But why not call `scrollToResource()` from the `useScroll` composable? IMO it would be de-coupled quite well and we would not need to add more complexity by adding more global events.

Reply:
@dschmidt Pushed commit, it seems to work quite well this way.

@JammingBen You mean calling useScroll  in usePagination? This one is not possible because useScroll comes from web-app-files and usePagination comes from web-pkg

Reply:
Ahh sorry, I thought it has been moved to `web-pkg`. Then this is fine by me!